crates/primitives/src/bits/address.rs

```diff
@@ -103,18 +102,41 @@ impl fmt::Display for Address {
 
 impl Address {
     /// Creates an Ethereum address from an EVM word's upper 20 bytes
-    /// (`hash[12..]`).
+    /// (`word[12..]`).
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use alloy_primitives::{address, b256, Address};
+    /// let word = b256!("000000000000000000000000d8da6bf26964af9d7eed9e03e53415d37aa96045");
+    /// assert_eq!(
+    ///     Address::from_word(word),
+    ///     address!("d8da6bf26964af9d7eed9e03e53415d37aa96045")
+    /// );
+    /// ```
     #[inline]
-    pub fn from_word(hash: FixedBytes<32>) -> Self {
-        Self(FixedBytes(hash[12..].try_into().unwrap()))
+    #[must_use]
+    pub fn from_word(word: FixedBytes<32>) -> Self {
+        Self(FixedBytes(word[12..].try_into().unwrap()))
```

Comment:
Consider for a future PR:
should this error if the top bytes are dirty?

Reply:
In another method maybe